### PR TITLE
AXI4-Lite crossbar implementation

### PR DESCRIPTION
**Pull request by WRoenninger**
Added an AXI4 LITE full crossbar implementation. It is structured the same way as the `axi_xbar` from an
instantiation perspective. The underlying `axi_lite_demux` and `axi_lite_mux` modules had to be specified separately as the LITE protocol does not need the elaborate switching decision handling as the full AXI4 protocol does. The three modules come with their respective documentation.
In `axi_test` a random AXI LITE master and slave class was added.
Fixes two issues: `axi_lite_to_axi` typos, `axi_decerror_slv` wrong parameter

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,9 +8,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## Unreleased
 
 ### Added
```
Comment (fabianschuiki) [inline CHANGELOG.md]: I was wondering if we should provide also `axi_lite_cut` and `axi_lite_multicut` modules to complement the AXI4-Lite story. What do you think?

File: doc/axi_lite_mux.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# AXI LITE Multiplexer
+
+The opposite function to the AXI LITE demultiplexer is performed by the AXI LITE Multiplexer. It merges multiple AXI-4 connections into one. The requests from multiple slave ports of the module get interleaved and transmitted over its one master port.
```
Comment (fabianschuiki) [inline doc/axi_lite_mux.md]:
```suggestion
The opposite function to the AXI-Lite demultiplexer is performed by the AXI-Lite multiplexer. It merges multiple AXI connections into one. The requests from multiple slave ports of the module get interleaved and transmitted over its one master port.
```

File: src/axi_decerr_slv.sv [hunk 3]
```diff
@@ -59,10 +59,10 @@ module axi_decerr_slv #(
...
   //--------------------------------------
   axi_atop_filter #(
-    .AXI_ID_WIDTH       ( AxiIdWidth ),
-    .AXI_MAX_WRITE_TXNS ( MaxTrans   ),
-    .req_t              ( req_t      ),
-    .resp_t             ( resp_t     )
+    .AxiIdWidth      ( AxiIdWidth ),
+    .AxiMaxWriteTxns ( MaxTrans   ),
+    .req_t           ( req_t      ),
+    .resp_t          ( resp_t     )
```
Comment (fabianschuiki) [inline src/axi_decerr_slv.sv]: Does this break other parts of the AXI repo? Seems unrelated to the AXI-Lite effort. Maybe we could split this out into a separate commit?

File: CHANGELOG.md (hunk 1 above)
Comment (zarubaf) [reply CHANGELOG.md]: That’s one of the advantages of using structs for the interfaces: We do not need a separate `_lite_cut` as the exakt type of the bus does not matter, as long as it provides `valid/ready` 😊

File: src/axi_decerr_slv.sv (hunk 3 above)
Comment (zarubaf) [reply src/axi_decerr_slv.sv]: Unfortunately we oversaw this at the previous release (`0.9.0`) which essentially broke it. I’ve pushed this as a hotfix to (`0.9.1`). The diff will go away once he rebased onto `0.9.1`.

Comment (fabianschuiki) [reply src/axi_decerr_slv.sv]: Awesome. 👍 for hot fixing.

File: CHANGELOG.md (hunk 1 above)
Comment (fabianschuiki) [reply CHANGELOG.md]: Completely forgot about that. That's great!

File: doc/axi_lite_demux.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+## Configuration
+
+This demultiplexer is configured through the parameters listed in the following table:
+
+| Name                 | Type               | Definition |
+|:---------------------|:-------------------|:-----------|
+| `NoMstPorts`         | `int unsigned`     | The number of AXI master ports of the demultiplexer (in other words, how many AXI slave modules can be attached). |
+| `MaxTrans`           | `int unsigned`     | The slave port can have at most this many transactions [in flight](../doc#in-flight). |
+| `FallThrough`        | `bit`              | Routing decisions on the AW channel fall through to the W channel.  Enabling this allows the demultiplexer to accept a W beat in the same cycle as the corresponding AW beat, but it increases the combinatorial path of the W channel with logic from `slv_aw_select_i`. |
+| `SpillXX`            | `bit`              | Inserts one spill register on the respective channel (AW, W, B, AR, and R) before the demultiplexer. |
```
[the remark this answers is not in the document]
Comment (WRoenninger) [reply doc/axi_lite_demux.md]: The first two culumns in each row have the same number of caracters, the last column is wraping around.

File: doc/axi_lite_demux.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+If all `SpillXX` and `FallThrough` are disabled, all paths through this multiplexer are combinatorial (i.e., have zero sequential latency).
+
+
+## Ports
+
+| Name                              | Description |
+|:----------------------------------|:------------|
+| `clk_i`                           | Clock to which all other signals (except `rst_ni`) are synchronous. |
+| `rst_ni`                          | Reset, asynchronous, active-low. |
+| `test_i`                          | Test mode enable (active-high). |
```
[the remark this answers is not in the document]
Comment (WRoenninger) [reply doc/axi_lite_demux.md]: is the connection to the testmode in fifo_v3

**Comment by accuminium** [comment]
Thanks for this contribution and the reviews!

I just rebased the branch on master to resolve the conflict with master and squash the fixup commits.

@zarubaf Do you agree with merging this PR?

**Comment by accuminium** [comment]
@WRoenninger: Could you please include the AXI-Lite crossbar in the synthesis bench and add that to this PR?

**Comment by zarubaf** [comment]
Sorry for the lack of response here. Yes, I am very happy with the contribution. Thanks a lot.

**Comment by WRoenninger** [comment]
Added the lite_xbar to the synth_bench, script elaborates

**Comment by accuminium** [comment]
Perfect, thanks. I could reproduce the behavioral TB and the synthesis bench.
